java/query/src/main/java/sleeper/query/model/output/S3ResultsOutput.java

```diff
@@ -70,25 +77,24 @@ public S3ResultsOutput(InstanceProperties instanceProperties, Schema schema, Map
         String defaultRowGroupSize = instanceProperties.get(DEFAULT_RESULTS_ROW_GROUP_SIZE);
         String defaultPageSize = instanceProperties.get(DEFAULT_RESULTS_PAGE_SIZE);
 
-        this.compressionCodec = config.getOrDefault(COMPRESSION_CODEC, "zstd");
         this.fileSystem = instanceProperties.get(FILE_SYSTEM);
         this.rowGroupSize = Long.parseLong(config.getOrDefault(ROW_GROUP_SIZE, defaultRowGroupSize));
         this.pageSize = Integer.parseInt(config.getOrDefault(PAGE_SIZE, defaultPageSize));
+        this.compressionCodec = tableProperties.get(sleeper.configuration.properties.table.TableProperty.COMPRESSION_CODEC);
+        this.dictionaryEncodingForRowKeyFields = tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_ROW_KEY_FIELDS);
+        this.dictionaryEncodingForSortKeyFields = tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_SORT_KEY_FIELDS);
+        this.dictionaryEncodingForValueFields = tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_VALUE_FIELDS);
     }
 
     @Override
     public ResultsOutputInfo publish(Query query, CloseableIterator<Record> results) {
         String outputFile = fileSystem + s3Bucket + "/query-" + query.getQueryId() + "/" + UUID.randomUUID() + ".parquet";
         ResultsOutputLocation outputLocation = new ResultsOutputLocation("s3", outputFile);
 
-        ParquetRecordWriter.Builder builder = new ParquetRecordWriter.Builder(new Path(outputFile),
-                SchemaConverter.getSchema(schema), schema)
-                .withCompressionCodec(CompressionCodecName.fromConf(compressionCodec))
-                .withRowGroupSize(rowGroupSize)
-                .withPageSize(pageSize);
         LOGGER.info("Opening writer for results of query {} to {}", query.getQueryId(), outputFile);
         long count = 0L;
-        try (ParquetWriter<Record> writer = builder.build()) {
+        try (ParquetWriter<Record> writer = ParquetRecordWriterFactory.createParquetRecordWriter(new Path(outputFile), schema, compressionCodec,
+            rowGroupSize, pageSize, dictionaryEncodingForRowKeyFields, dictionaryEncodingForSortKeyFields, dictionaryEncodingForValueFields, new Configuration())) {
```

Comment:
We could get a builder for the writer based on the table properties, and add some logic to override the fields that are set in the config map.

Reply:
This can be resolved as part of https://github.com/gchq/sleeper/issues/686